storops/unity/resource/snap.py

```diff
@@ -97,22 +97,24 @@ def existed(self):
                 self.state != SnapStateEnum.DESTROYING)
 
     @version('>=4.1')
-    def attach_to(self, host, access_mask=SnapAccessLevelEnum.READ_WRITE):
-        host_access = [{'host': host, 'allowedAccess': access_mask}]
-        # If this lun has been attached to other host, don't overwrite it.
+    def attach_to(self, *hosts, access_mask=SnapAccessLevelEnum.READ_WRITE):
+        host_access = [{'host': host, 'allowedAccess': access_mask} for host in hosts]
+
+        # Snapshot access cannot be updated, must be detached then set anew
         if self.host_access:
-            host_access += [{'host': item.host,
-                             'allowedAccess': item.allowed_access}
-                            for item in filter(lambda x: x.host.id == host.id,
-                                               self.host_access)]
+            self.detach_from()
```

Comment:
Attaching to new hosts detachs the old hosts is not acceptable, it causes IO failure of original hosts.

Reply:
Unfortunately unless you know of a hidden/undocumented way to modify snapshot host_access, it seems to simply not be possible, as discussed in #205. I think this is actually ok however, as per [Unity snapshot doco, p10](https://support.emc.com/docu69887_Dell-EMC-Unity:-Snapshots-and-Thin-Clones---A-Detailed-Review.pdf?language=en_US), we can just create additional snapshots and attach those to different (or same) hosts as required.

That said, maybe it would in fact be better to raise exception if `attach_to` is called after access is already configured, forcing the user to run `detach_from` first if they want to modify access. This would then match the Unisphere experience, and ensure storops can't be blamed for any resulting IO failure. I'm happy to remove the call to `self.detach_from()` if this is your preference.

Reply:
Did you try my suggestion below?
>To add new host, you can use the modify action to accomplish what you want.